### PR TITLE
feat: add support for undeleting messages

### PR DESCRIPTION
**Pull request by akupila**
# Submit a pull request

## CLA

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

This adds support for undeleting messages that were previously soft-deleted

### CODE REVIEW

**Review by kanat — APPROVED** [review]
LGTM

File: stream_chat/client.py [hunk 1]
```diff
@@ -321,6 +321,11 @@ def update_message_partial(
     def delete_message(self, message_id: str, **options: Any) -> StreamResponse:
         return self.delete(f"messages/{message_id}", options)
 
+    def undelete_message(self, message_id: str, user_id: str) -> StreamResponse:
+        return self.post(
+            f"messages/{message_id}/undelete", data={"undeleted_by": user_id}
+        )
```
Comment (kanat) [inline stream_chat/client.py:327]: Should we add this to async client as well? :)

Comment (akupila) [reply stream_chat/client.py:328]: Good one. I updated it now 👍

**Comment by akupila** [comment]
@gumuz friendly ping 🙂

**Review by gumuz — APPROVED** [review]

**Pull request merged**